_benchmark/workloads/vectorsearch.md

```diff
@@ -0,0 +1,569 @@
+| `query_data_set_format`                   | The format of the vector data set used for queries. Only one of `query_k`, `query_max_distance`, and, `query_min_score` can be provided.        |
+| `query_data_set_path`                     | The path to the vector data set used for queries.                                                                                               |
+| `query_count`                             | The number of queries for the search operation.                                                                                                 |
+| `query_body`                              | The JSON properties that will be merged with search body                                                                                        |
+| `search_clients`                          | The number of clients to use for running queries.                                                                                               |
+| `repetitions`                             | THe number of repetitions until the data set is exhausted. Default is `1`.                                                                      |
+| `target_throughput`                       | The target throughput for each query operation in requests per second. Default is `10`.                                                         |
+| `time_period`                             | The period of time dedicated for the run the benchmark test in seconds. Default is `900`.                                                       |
+
+
+
+## Test procedures
+
+The workload supports the following test procedures.
+
+### No Train Test
+
+The No Train Test procedure tests vector search indices that require no training. You can define the underlying configuration of the vector search algorithm (such as specific engine, space type, etc.) as method definitions. Check vector search method definitions for more details.
+
+### No Train Test Index Only
+
+Th No Train Test Index Only procedure is used to index vector search indices that require no training. It's particularly useful when you want to benchmark only the indexing operation.
+
+### Force Merge Index
+
+The Force Merge Index procedure optimizes vector search indices by performing force merge operations up to a given maximum number of segments. For large datasets, force merge is a costly operation. Therefore, it's better to have a separate procedure to trigger force merge operations occasionally based on user requirements.
+
+### Search
+
+The Search procedure benchmarks previously indexed vector search indices. It's useful when you want to benchmark large vector search indices without re-indexing each time, since load time can be substantial for large datasets. This procedure includes warmup operations to avoid cold start problems during vector search.
```

Comment:
```suggestion
The Search procedure benchmarks previously indexed vector search indexes. It's useful when you want to benchmark large vector search indices without re-indexing each time, since load time can be substantial for large datasets. This procedure includes warmup operations to avoid cold start problems during vector search.
```